### PR TITLE
MRG: Deprecations

### LINKED ISSUE #3623: BUG: sklearn cannot find LeaveOneLabelOut
On latest `sklearn` I get this error in `master`:
```
======================================================================
ERROR: Test time generalization decoding
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/larsoner/custombuilds/nose/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/larsoner/custombuilds/mne-python/mne/utils.py", line 807, in dec
    return function(*args, **kwargs)
  File "/home/larsoner/custombuilds/mne-python/mne/decoding/tests/test_time_gen.py", line 61, in test_generalization_across_time
    from sklearn.model_selection import (KFold, StratifiedKFold,
ImportError: cannot import name LeaveOneLabelOut
```
@kingjr @agramfort were we using a deprecated feature, or is this a sklearn bug?

### PR DESCRIPTION
**Pull request by Eric89GXL**
Doing my part for release :)

This is all the deprecations for 0.14.

Ready for review/merge from my end.

Closes #3623.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
CIs are happy (or were on last run before I push-forced to get them all to run on the same commit)

**Comment by agramfort** [comment]
I will let @jaeilepp take a first look when fulling done with group demo analysis

File: mne/tests/test_epochs.py [hunk 1]
```diff
@@ -101,23 +101,22 @@ def test_average_movements():
...
     crop = 0., 10.
     origin = (0., 0., 0.04)
-    raw = read_raw_fif(fname_raw_move, allow_maxshield='yes',
-                       add_eeg_ref=False)
+    raw = read_raw_fif(fname_raw_move, allow_maxshield='yes')
     raw.info['bads'] += ['MEG2443']  # mark some bad MEG channel
-    raw.crop(*crop, copy=False).load_data()
-    raw.filter(None, 20, method='iir')
+    raw.crop(*crop).load_data()
+    raw.filter(None, 20)
```
Comment (Eric89GXL) [inline mne/tests/test_epochs.py]: No need to filter with IIR, and the test with IIR mode doesn't actually work on old versions of SciPy. Easier to make it use FIR than to add a decorator (and decrease coverage on those runs)

**Comment by Eric89GXL** [comment]
Comments addressed, thanks @jaeilepp

File: mne/preprocessing/stim.py [hunk 2]
```diff
@@ -44,7 +43,7 @@ def _fix_artifact(data, window, picks, first_samp, last_samp, mode):
 
 
 def fix_stim_artifact(inst, events=None, event_id=None, tmin=0.,
-                      tmax=0.01, mode='linear', stim_channel=None, copy=None):
+                      tmax=0.01, mode='linear', stim_channel=None):
```
Comment (agramfort) [inline mne/preprocessing/stim.py]: this was done by warning in 0.13 release

File: examples/decoding/plot_linear_model_patterns.py [hunk 3]
```diff
@@ -45,7 +46,7 @@
 
 # Setup for reading the raw data
 raw = io.read_raw_fif(raw_fname, preload=True)
-raw.filter(2, None, method='iir')  # replace baselining with high-pass
+raw.filter(2, 15)  # replace baselining with high-pass
```
Comment (Eric89GXL) [inline examples/decoding/plot_linear_model_patterns.py]: It fixes a warning that this example generates about deprecation causing aliasing

**Comment by Eric89GXL** [comment]
@agramfort fixed the `n_jobs=1`. Okay to merge once the CIs come back happy?

**Comment by agramfort** [comment]
Yes ok






On Sun, Oct 2, 2016 at 12:33 PM +0900, "Eric Larson" <notifications@github.com> wrote:












@agramfort fixed the n_jobs=1. Okay to merge once the CIs come back happy?

**Pull request merged**